lib/Auth.js

```diff
@@ -249,27 +298,46 @@ class Auth {
     //
     //
     isValidAccessToken(accessToken) {
-        if (this._auth.accessToken === accessToken) {
-            return true;
-        } else {
+        return this.getuserForAccessToken(accessToken) !== null;
+    }
+    //
+    //
+    //
+    getuserForAccessToken(accessToken) {
+        const accessTokenInfo = this._auth.accessTokens[accessToken];
+        if (typeof accessTokenInfo === 'undefined') {
+            this.debug('Auth:isValidAccessToken(): accessToken = ' + JSON.stringify(accessToken) + ' not found.');
+            return null;
+        }
+        const user = accessTokenInfo.user;
+        const expiresAt = new Date(accessTokenInfo.expiresAt);
+        if (expiresAt < new Date()) {
             this.debug('Auth:isValidAccessToken(): accessToken = ' + JSON.stringify(accessToken));
-            this.debug('Auth:isValidAccessToken(): this._auth = ' + JSON.stringify(this._auth));
-            this.debug('Auth:isValidAccessToken(): invalid accessToken');
-            return false;
+            this.debug('Auth:isValidAccessToken(): user = ' + user);
+            this.debug('Auth:isValidAccessToken(): expiresAt = ' + expiresAt.toLocaleString());
+            this.debug('Auth:isValidAccessToken(): accessToken expired');
+            return null;
         }
+        return user;
     }
     //
     //
     //
     isValidRefreshToken(refreshToken) {
-        if (this._auth.refreshToken === refreshToken) {
-            return true;
-        } else {
-            this.debug('Auth:isValidRefreshToken(): refreshToken = ' + JSON.stringify(refreshToken));
-            this.debug('Auth:isValidRefreshToken(): this._auth = ' + JSON.stringify(this._auth));
-            this.debug('Auth:isValidRefreshToken(): invalid refreshToken');
+        const refreshTokenInfo = this._auth.refreshTokens[refreshToken];
+        if (typeof refreshTokenInfo === 'undefined') {
+            this.debug('Auth:isValidRefreshToken(): refreshToken not found ' + JSON.stringify(accessToken));
```

Comment:
`JSON.stringify(accessToken))` ... accessToken is not defined here.

Reply:
I'm sorry about that. I've fixed it.